aiomysql/pool.py

```diff
@@ -152,6 +152,12 @@ async def _fill_free_pool(self, override_min):
                 self._free.pop()
                 conn.close()
 
+            # On MySQL 8.0 a timed out connection sends an error packet before
+            # closing the connection, preventing us from relying on at_eof().
+            elif conn._reader.eof_received:
```

Comment:
this does not exist on a unix socket reader and therefore currently breaks unix sockets.

Reply:
based on minimal local testing it seems the behavior on unix sockets is the same as on tcp connections, where `StreamReader._eof` is `True` on such connections.